### ISSUE TITLE
Focus-visible not working correctly on input fields

### ISSUE THREAD

**Issue by grandorf** [youseedk/dna#279]
Apparently the focus-visible technique is not working properly (I used chrome).

Expected result would be to only show the focus state on tabbing with keyboard. If I focus using the mouse it is not necessary to add the blue outline. But it does.

This seems to work for checkbox and radio buttons so there must be a bug in CSS.

![image](https://user-images.githubusercontent.com/21218852/98835834-36745580-2441-11eb-8853-33e457e2d9ab.png)

**Comment by Lubbe91** [comment]
https://github.com/WICG/focus-visible/issues/131#issuecomment-375725010

**Comment by rtatarinov4xxi** [comment]
I agree with author of issue. This issue is duplicate for this: https://github.com/WICG/focus-visible/issues/131
However, I think it will be customize for user. I, like an user need to able to enable focus by tab only, without click, even if default behavior is a focus-visible when focus in input field.

**Comment by MBehtemam** [comment]
also I need this, currently I disabled it with `!important` as it's not a good idea.

**Comment by havgry** [comment]
Hmm, is this a regression? I haven't checked previous versions, but I don't remember this behaviour. Wasn't outline disabled altogether?

**Comment by Dechowmedia** [comment]
Still an issue to this day :)
![image](https://user-images.githubusercontent.com/816899/106107569-5f0a1280-6147-11eb-95e5-aeec8db3224d.png)

I think maybe we should heavily consider utilizing :focus instead of :focus-visible - as we do indicate the focus / state change with our own border - outline.

I don't believe having competing colors and focus states will aid this - I would actually recommend us utilizing the background color of the input field, as else we will have multiple fighting borders.

**Comment by maciej-panecki-wttech** [comment]
This is not an issue. It seems counter intuitive but according to specs: https://www.w3.org/TR/selectors-4/#the-focus-visible-pseudo 

> Any element which supports keyboard input (such as an input element, or any other element which may trigger a virtual keyboard to be shown on focus if a physical keyboard is not present) should always match :focus-visible when focused.
